### PR TITLE
Automatically publish SNAPSHOTs on pushes to master

### PR DESCRIPTION
**Pull request by jackkoenig**
Uses [sbt-ci-release](https://github.com/olafurpg/sbt-ci-release) for automation. Any push to the specific branches (ie. a CI run _after_ a PR is merged)  will push SNAPSHOTs to the Sonatype snapshot repository. This is set up for master and all stable branches (See the regex: `/^\d+\.\d+\.x$/`)

The affected branches:
| branch | SNAPSHOT |
| --- | --- |
| `master` | `1.5-SNAPSHOT` |
| `1.4.x` | `1.4-SNAPSHOT` |
| `1.3.x` | `1.3-SNAPSHOT` |
| `1.2.x` | `1.2-SNAPSHOT` |

Backports will likely conflict and need fixup.

### Contributor Checklist

- [NA] Did you add Scaladoc to every public function/method?
- [NA] Did you update the FIRRTL spec to include every new feature/behavior?
- [😬 ] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [NA] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

  - automation

#### API Impact

No impact

#### Backend Code Generation Impact

No impact

#### Desired Merge Strategy

- Squash

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (1.2.x, 1.3.0, 1.4.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Comment by jackkoenig** [comment]
It worked! `1.5-SNAPSHOT` was pushed for 2.11, 2.12, and 2.13 by this PR (Not the commented out line in the `.travis.yml`, normally only pushes should do the publishing).

https://oss.sonatype.org/#nexus-search;gav~edu.berkeley.cs~firrtl_2.12~~~~kw,versionexpand

File: .travis.yml [hunk 1]
```diff
@@ -25,8 +30,10 @@ before_script:
   - if [ $OLDEST_SHARED == $OLDEST_COMMIT ]; then git fetch --unshallow; fi
 
 stages:
-  - prepare
-  - test
+  - name: prepare
+  - name: test
+  - name: release
+    if: ((branch = master) OR (branch =~ /^\d+\.\d+\.x$/)) AND (type = push) AND (NOT fork)
```
Comment (chick) [inline .travis.yml:36]: I'm assuming this is the correct equals here. It's a syntax diversity meeting here

**Review by chick — APPROVED** [review]
That looks like a nice small change. All looks ok to me.
See the comment about the comparsion operators in .travis.yml:36 file

File: .travis.yml (hunk 1 above)
Comment (jackkoenig) [reply .travis.yml:36]: I think it's right but we shall see, got this from https://docs.travis-ci.com/user/conditions-v1

**Pull request merged**